File: frontend/feature/history/src/main/kotlin/com/matrigluco/feature/history/data/remote/HistoryApi.kt
package com.matrigluco.feature.history.data.remote

import com.matrigluco.feature.history.data.remote.dto.HistoryListResponseDto
import retrofit2.http.GET
import retrofit2.http.Query

interface HistoryApi {

    @GET("api/v1/history")
    suspend fun getHistory(
        @Query("types") types: String? = null,
        @Query("date_from") dateFrom: String? = null,
        @Query("date_to") dateTo: String? = null,
        @Query("month") month: String? = null,
        @Query("page") page: Int = 1,
        @Query("page_size") pageSize: Int = 50
    ): HistoryListResponseDto
}

File: web/src/features/history/api/history.api.ts
import { apiRequest } from "../../../services/http/client";
import { HistoryListDto } from "../types/history.types";

export interface GetHistoryParams {
  types?: string; // Comma separated: "assessment,measurement,report,consultation"
  date_from?: string;
  date_to?: string;
  month?: string; // "YYYY-MM"
  page?: number;
  page_size?: number;
}

export const historyApi = {
  async getHistory(
    params?: GetHistoryParams,
    signal?: AbortSignal
  ): Promise<HistoryListDto> {
    return apiRequest<HistoryListDto>({
      url: "/history",
      method: "GET",
      params,
      signal,
    });
  },
};
